gnovm/pkg/doc/json_doc.go

```diff
@@ -0,0 +1,142 @@
+type JSONDocumentation struct {
+	PackagePath string `json:"package_path"`
+	PackageLine string `json:"package_line"` // package io // import "io"
+	PackageDoc  string `json:"package_doc"`  // markdown of top-level package documentation
+	// https://pkg.go.dev/go/doc#Package.Markdown to render markdown
+
+	// These match each of the sections in a pkg.go.dev package documentationj
+	Values []*JSONValue `json:"values"` // constants and variables declared
+	Funcs  []*JSONFunc  `json:"funcs"`  // Funcs and methods
+	Types  []*JSONType  `json:"types"`
+}
+
+type JSONValue struct {
+	Signature string `json:"signature"`
+	Doc       string `json:"doc"` // markdown
+}
+
+type JSONFunc struct {
+	Type      string `json:"type"` // if this is a method
+	Name      string `json:"name"`
+	Signature string `json:"signature"`
+	Doc       string `json:"doc"` // markdown
+}
+
+type JSONType struct {
+	Name      string `json:"name"`
+	Signature string `json:"signature"`
+	Doc       string `json:"doc"` // markdown
+}
+
+// NewDocumentableFromMemPkg gets the pkgData from memPkg and returns a Documentable
+func NewDocumentableFromMemPkg(memPkg *gnovm.MemPackage, unexported bool, symbol, accessible string) (*Documentable, error) {
+	pd, err := newPkgDataFromMemPkg(memPkg, unexported)
+	if err != nil {
+		return nil, err
+	}
+
+	doc := &Documentable{
+		bfsDir:     pd.dir,
+		pkgData:    pd,
+		symbol:     symbol,
+		accessible: accessible,
+	}
+	return doc, nil
+}
+
+// WriteJSONDocumentation returns a JSONDocumentation for the package
+func (d *Documentable) WriteJSONDocumentation() (*JSONDocumentation, error) {
+	opt := &WriteDocumentationOptions{}
+	_, pkg, err := d.pkgData.docPackage(opt)
+	if err != nil {
+		return nil, err
+	}
+
+	jsonDoc := &JSONDocumentation{
+		PackagePath: d.pkgData.dir.dir,
+		PackageLine: fmt.Sprintf("package %s // import %q", pkg.Name, pkg.ImportPath),
+		PackageDoc:  string(pkg.Markdown(pkg.Doc)),
+		Values:      []*JSONValue{},
+		Funcs:       []*JSONFunc{},
+		Types:       []*JSONType{},
+	}
+
+	for _, value := range pkg.Consts {
+		buf := new(strings.Builder)
+		if err := format.Node(buf, d.pkgData.fset, value.Decl); err != nil {
+			return nil, err
+		}
+		jsonDoc.Values = append(jsonDoc.Values, &JSONValue{
+			Signature: buf.String(),
+			Doc:       string(pkg.Markdown(value.Doc)),
+		})
+	}
+
+	for _, value := range pkg.Vars {
+		buf := new(strings.Builder)
+		if err := format.Node(buf, d.pkgData.fset, value.Decl); err != nil {
+			return nil, err
+		}
+		jsonDoc.Values = append(jsonDoc.Values, &JSONValue{
+			Signature: buf.String(),
+			Doc:       string(pkg.Markdown(value.Doc)),
+		})
+	}
+
+	for _, fun := range pkg.Funcs {
+		buf := new(strings.Builder)
+		if err := format.Node(buf, d.pkgData.fset, fun.Decl); err != nil {
+			return nil, err
+		}
+		jsonDoc.Funcs = append(jsonDoc.Funcs, &JSONFunc{
+			Name:      fun.Name,
+			Signature: buf.String(),
+			Doc:       string(pkg.Markdown(fun.Doc)),
```

Comment:
```suggestion
			Doc:       string(pkg.Markdown(fun.Doc)),
			Params: []JSONField{...},
			Results: []JSONField{...},
```

JSONField is a simple `struct{ Name string; Type string }`, breaking down the `ast.Field` type